File: app/src/main/java/com/nicholasdoglio/eyebleach/model/reddit/ChildData.java
package com.nicholasdoglio.eyebleach.model.reddit;

import com.squareup.moshi.Json;


public class ChildData {

    @Json(name = "selftext")
    private String selftext;
    @Json(name = "thumbnail")
    private String thumbnail;
    @Json(name = "permalink")
    private String permalink;
    @Json(name = "url")
    private String url;
    @Json(name = "id")
    private String id;

    public String getSelftext() {
        return selftext;
    }

    public void setSelftext(String selftext) {
        this.selftext = selftext;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }

    public String getPermalink() {
        return permalink;
    }

    public void setPermalink(String permalink) {
        this.permalink = permalink;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}

File: app/src/main/java/com/nicholasdoglio/eyebleach/data/model/reddit/ChildData.kt
/*
    Aww Gallery
    Copyright (C) 2017  Nicholas Doglio

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.nicholasdoglio.eyebleach.data.model.reddit

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.squareup.moshi.Json

/**
 * @author Nicholas Doglio
 */
@Entity
class ChildData {
    @Json(name = "selftext")
    var selftext: String? = null
    @Json(name = "thumbnail")
    var thumbnail: String? = null
    @Json(name = "permalink")
    var permalink: String? = null
    @Json(name = "url")
    var url: String? = null
    @Json(name = "id")
    var id: String? = null
    @PrimaryKey(autoGenerate = true)
    var key: Int? = null



    fun fullUrl() = "https://reddit.com$permalink"
}
